pages/index.js

```diff
@@ -1,32 +1,227 @@
+import { useEffect, useState, useRef } from 'react'
 import Head from 'next/head'
-import Image from 'next/image'
+import 'mapbox-gl/dist/mapbox-gl.css'
+import 'react-spring-bottom-sheet/dist/style.css'
+import { BottomSheet } from 'react-spring-bottom-sheet'
+
+import mapboxgl from '!mapbox-gl'
 import styles from '../styles/Home.module.css'
 
+import { provincesWithCities } from './api/bed'
+
+mapboxgl.accessToken = process.env.NEXT_PUBLIC_MAPBOX
+
 export default function Home() {
+  const mapContainer = useRef(null)
+  const map = useRef(null)
+  const [lng, setLng] = useState(115.212631)
+  const [lat, setLat] = useState(-8.670458)
+  const [zoom, setZoom] = useState(9)
+
+  const [city, setCity] = useState({ value: 'bali', label: 'Bali' })
+  const [hospitals, setHospitals] = useState([])
+  const [popupCityVisible, setPopupCityVisibility] = useState(false)
+  const [popupHospital, setPopupHospitalVisibility] = useState(false)
+
+  useEffect(() => {
+    if (map.current) return // initialize map only once
+    map.current = new mapboxgl.Map({
+      container: mapContainer.current,
+      style: 'mapbox://styles/mapbox/streets-v11',
+      center: [lng, lat],
+      zoom: zoom,
+    })
+
+    map.current.on('load', function () {
+      map.current.resize()
+    })
+  })
+
+  useEffect(() => {
+    fetchHospital()
+  }, [city])
+
+  useEffect(() => {
+    updateMap()
+  }, [hospitals])
+
+  const fetchHospital = () => {
+    fetch(`/api/bed?prov=${city.value}`)
```

Comment:
Mind to replace this with `axios`? It's already installed so you can use it directly.